### PR TITLE
feat: add OOG error handling for `CALLCODE`, `DELEGATECALL` and `STATICCALL` to `ErrorOOGCall`

### PR DESCRIPTION
**Pull request by silathdiir**
Upstream spec issue https://github.com/privacy-scaling-explorations/zkevm-specs/issues/360

Spec PR https://github.com/scroll-tech/zkevm-specs/pull/56

### Summary

1. Merge enum value `OogError::CallCode`, `DelegateCall` and `StaticCall` into `OogError::Call`.

2. Merge enum value `ExecutionState::ErrorOutOfGasCALLCODE`, `ErrorOutOfGasDELEGATECALL` and `ErrorOutOfGasSTATICCALL` into `ExecutionState::ErrorOutOfGasCALL`.

3. Update bus-mapping and zkevm-circuit `error_oog_call` to handle CALLCODE, DELEGATECALL and STATICCALL opcodes.

4. Update both root and internal test cases in circuit.

### CODE REVIEW

File: zkevm-circuits/src/evm_circuit/execution/error_oog_call.rs [hunk 1]
```diff
@@ -240,27 +314,35 @@ mod test {
...
+        bytecode.append(&bytecode! {
+            PUSH32(address.to_word())
             PUSH32(Word::from(stack.gas))
-            CALL
+            .write_op(opcode)
             PUSH1(0)
             PUSH1(0)
-            .write_op(terminator)
-        };
+            .write_op(if is_success { OpcodeId::RETURN } else { OpcodeId::REVERT })
```
Comment (DreamWuGit) [inline zkevm-circuits/src/evm_circuit/execution/error_oog_call.rs:338]: I think don't need `is_success` check since it is an error case.

**Comment by DreamWuGit** [comment]
overal good to me .

File: zkevm-circuits/src/evm_circuit/execution/error_oog_call.rs (hunk 1 above)
Comment (silathdiir) [reply zkevm-circuits/src/evm_circuit/execution/error_oog_call.rs]: Deleted `is_success` parameter and always use `OpcodeId::REVERT` in test cases of OOG call.
Fixed in https://github.com/scroll-tech/zkevm-circuits/pull/262/commits/a597c2c92b44dc45bf0e9760914243fb91742259

**Comment by silathdiir** [comment]
Thanks for @DreamWuGit 's help for code review. Moved to upstream https://github.com/privacy-scaling-explorations/zkevm-circuits/pull/1127

**Pull request closed without merging**